pkg/storaged/smart-details.jsx

```diff
@@ -0,0 +1,120 @@
+
+import { Card, CardBody, CardHeader, CardTitle, DescriptionList, DescriptionListDescription, DescriptionListGroup, DescriptionListTerm } from "@patternfly/react-core";
+import { Dropdown, DropdownItem, KebabToggle } from '@patternfly/react-core/dist/esm/deprecated/components/Dropdown/index.js';
+
+const _ = cockpit.gettext;
+
+const selftestStatusDescription = {
+    success: _("Successful"),
+    aborted: _("Aborted"),
+    interrupted: _("Interrupted"),
+    fatal: _("Did not complete"),
+    error_unknown: _("Failed (Unknown)"),
+    error_electrical: _("Failed (Electrical)"),
+    error_servo: _("Failed (Servo)"),
+    error_read: _("Failed (Read)"),
+    error_handling: _("Failed (Damaged)"),
+    inprogress: _("In progress"),
+};
+
+const SmartActions = ({ smartInfo }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+    const smartSelftestStatus = smartInfo.SmartSelftestStatus;
+
+    const runSmartTest = async (type) => {
+        await smartInfo.SmartSelftestStart(type, {});
+    };
+
+    const abortSmartTest = async () => {
+        await smartInfo.SmartSelftestAbort({});
+    };
+
+    const actions = [
+        <DropdownItem key="run-short-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('short') }}>
+            {_("Run short test")}
+        </DropdownItem>,
+        <DropdownItem key="run-extended-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('extended') }}>
+            {_("Run extended test")}
+        </DropdownItem>,
+        <DropdownItem key="run-conveyance-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSmartTest('conveyance') }}>
+            {_("Run conveyance test")}
+        </DropdownItem>,
+    ];
+
+    if (smartInfo.SmartSelftestStatus === "inprogress") {
+        actions.push(
+            <DropdownItem key="abort-smart-test"
+                          onClick={() => { setKebabOpen(false); abortSmartTest('conveyance') }}>
+                {_("Abort test")}
+            </DropdownItem>,
+        );
+    }
+
+    return (
+        <Dropdown toggle={<KebabToggle onToggle={(_, isOpen) => setKebabOpen(isOpen)} />}
+                isPlain
+                isOpen={isKebabOpen}
+                position="right"
+                id="smart-actions"
+                dropdownItems={actions} />
+    );
+};
+
+export const SmartDetails = ({ smartInfo }) => {
+    const SmartDetailRow = ({ title, value }) => {
+        if (value === undefined)
+            return null;
+
+        return (
+            <DescriptionListGroup>
+                <DescriptionListTerm>{title}</DescriptionListTerm>
+                <DescriptionListDescription>{value}</DescriptionListDescription>
+            </DescriptionListGroup>
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19103-20230717-115949-00eb709f-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)